File: CoWorkHub/CoWorkHub.Model/Role.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoWorkHub.Model
{
    public class Role
    {
        public int RolesId { get; set; }
        public string RoleName { get; set; } = null!;
        public string? Description { get; set; }
        public bool? IsDeleted { get; set; } = false;
    }
}

File: CoWorkHub/UI/coworkhub_desktop/lib/models/role.dart
import 'package:json_annotation/json_annotation.dart';

part 'role.g.dart';

@JsonSerializable()
class Role {
  int rolesId;
  String roleName;
  String? description;
  bool? isDeleted;

  Role({
    required this.rolesId,
    required this.roleName,
    this.description,
    this.isDeleted = false,
  });

  factory Role.fromJson(Map<String, dynamic> json) => _$RoleFromJson(json);
  Map<String, dynamic> toJson() => _$RoleToJson(this);
}
